package/MDAnalysis/core/selection.py

```diff
@@ -794,7 +794,73 @@ def _sel_with_icodes(self, vals, icodes):
         return mask
 
 
+class BoolSelection(Selection):
+    """Range selection for boolean values"""
+
+    def __init__(self, parser, tokens):
+        values = grab_not_keywords(tokens)
+        if not values:
+            values = [True]
+        val = values[0]
+        if isinstance(val, str):
+            val = val.lower()
+            if val == "false":
+                val = False
+            elif val == "true":
+                val = True
+        self.value = val
+
+    def apply(self, group):
+        vals = getattr(group, self.field)
+        return group[vals == self.value].unique
+
+
+class FloatRangeSelection(Selection):
+    """Range selection for float values"""
+
+    def __init__(self, parser, tokens):
+        values = grab_not_keywords(tokens)
+        if not values:
+            raise ValueError("Unexpected token: '{0}'".format(tokens[0]))
+
+        uppers = []  # upper limit on any range
+        lowers = []  # lower limit on any range
+
+        for val in values:
+            try:
+                lower = float(val)
+                upper = None
+            except ValueError:
+                # check if in appropriate format 'lower:upper' or 'lower-upper'
+                selrange = re.match(r"(\d+\.?\d*)[:-](\d+\.?\d*)", val)
+                if not selrange:
+                    errmsg = f"Failed to parse number: {val}"
+                    raise ValueError(errmsg) from None
+                lower, upper = map(float, selrange.groups())
+
+            lowers.append(lower)
+            uppers.append(upper)
+
+        self.lowers = lowers
+        self.uppers = uppers
+
+    def apply(self, group):
+        mask = np.zeros(len(group), dtype=bool)
+        vals = getattr(group, self.field)
+
+        for upper, lower in zip(self.uppers, self.lowers):
+            if upper is not None:
+                thismask = vals >= lower
+                thismask &= vals <= upper
+            else:
+                thismask = vals == lower
```

Comment:
Equivalence is iffy for floats. I'd be almost inclined to raise an error and tell users that they need to do a range for floats. However, I also see how users will get annoyed when "val == 1" works and "val = 1.0" raises an exception.

So maybe just document that equality is problematic...